### PR TITLE
Route /photos

### PR DESCRIPTION
**Pull request by vgeorge**
See #7. This is a PR should cover GET/POST/PACTH/DEL methods for photos.

Method POST is implemented. It requires a photo as base64 string, `lon`, `lat`, `bearing` and `createdAt` attributes. It adds the photo to the database, includes coordinates and creation date in the EXIF and writes file in different resolutions, which can be accessed via `/media` route.

### CODE REVIEW

**Comment by vgeorge** [comment]
All /photos methods implemented, this is ready for review.

@batpad, there is a lot of commits here, I suggest checking the test file to confirm it covers expected use cases and check if docs are ok.

File: app/services/media-store.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      GPSDestBearing: bearing
+    });
+
+    // Write resized files
+    for (let i = 0; i < sizes.length; i++) {
+      const { width, height, id } = sizes[i];
+      const resizedFilePath = `${baseFilePath}-${id}.jpg`;
+      await sharp(originalFilePath)
+        .withMetadata()
+        .resize(width, height, { fit: 'inside' })
```
Comment (batpad) [inline app/services/media-store.js]: @vgeorge let's discuss if `fit`: `inside` is the best, or another option? I'd personally vote for keeping width constant and let height be as per aspect ratio, but happy to discuss.

File: app/routes/photos/get.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { getPhoto, photoToJson } from '../../models/photos';
+import logger from '../../services/logger';
+
+const idLength = config.get('idLength');
+
+/**
+ * @apiGroup Photos
+ *
+ * @api {get} /photos/:id  4. GET /photos/:id
+ * @apiDescription Get photo, must be owner or admin.
```
Comment (batpad) [inline app/routes/photos/get.js]: Do we need to restrict GET calls for photos to owners / admins? @vgeorge

File: app/services/media-store.js (hunk 1 above)
Comment (vgeorge) [reply app/services/media-store.js]: I would leave as is for now and discuss in a separate issue.

File: app/routes/photos/get.js (hunk 2 above)
Comment (vgeorge) [reply app/routes/photos/get.js]: In fact, the only restriction now is being a logged user, so everyone can we all photos. Let's clarify in a separated issue what is the proper data strategy.

**Pull request merged**